src/intel/RoomIntel.ts

```diff
@@ -456,7 +458,46 @@ export class RoomIntel {
 			}
 		}
 	}
+	private static autoPoison(room: Room) {
+		
+		if(!Memory.settings.autoPoison.enabled || !room || getAllColonies().length == Game.gcl.level) {
+			return;
+		}
 
+		// poison X rooms at a time
+		let count = 0;
+		for (const colony of getAllColonies()) {
+			if(Game.flags['POISON-'+colony.name]) {
+				count++;
+			}
+			if (count >= Memory.settings.autoPoison.concurrent) {
+				return;
+			}
+		}
+		
+		if (!(room.controller && room.controller.level == 0 && room.controller.reservation == undefined 
+			&& !room.storage && !room.terminal // do not include rooms with storage/terminal, i might want to loot
```

Comment:
I'd consider adding a static method in DirectivePoison that is "I can poison this room" that does the walkable positions thing too.

Reply:
Agreed